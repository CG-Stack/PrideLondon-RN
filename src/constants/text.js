export default {
  tabEvents: "Events",
  filterByInterest: "I'm interested in...",
  eventDetailsPrice: "From: Â£",
  eventDetailsAccessibility: "Accessibility",
  eventDetailsAccessibilityDetails: "Accessibility Details",
  eventDetailsContact: "Contact",
  eventDetailsBuyButton: "Buy Tickets",
  eventDetailsGenderNeutralToilets: "Gender neutral toilets"
};
