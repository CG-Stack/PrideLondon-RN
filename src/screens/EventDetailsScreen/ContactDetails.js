// @flow
import React from "react";
import { Image, StyleSheet } from "react-native";
import { email as sendEmail, phonecall } from "react-native-communications";
import LayoutColumn from "../../components/LayoutColumn";
import Text from "../../components/Text";
import IconItem from "./IconItem";
import emailIcon from "../../../assets/images/email.png";
import callIcon from "../../../assets/images/call.png";
import text from "../../constants/text";
import TextLink from "../../components/TextLink";

type Props = {
  email: string,
  phone: string
};

const ContactDetails = ({ email, phone }: Props) => (
  <LayoutColumn spacing={4}>
    <Text type="h2" color="lightNavyBlueColor">
      {text.eventDetailsContact}
    </Text>
    <LayoutColumn spacing={8}>
      {email && (
        <IconItem icon={<Image source={emailIcon} style={styles.mailIcon} />}>
          <TextLink
            onPress={() =>
              sendEmail(
                [email],
                null,
                null,
                text.eventDetailsContactEmailSubject,
                text.eventDetailsContactEmailBody
              )
            }
          >
            {email}
          </TextLink>
        </IconItem>
      )}
      {phone && (
        <IconItem icon={<Image source={callIcon} style={styles.phoneIcon} />}>
          <TextLink onPress={() => phonecall(phone, false)}>{phone}</TextLink>
        </IconItem>
      )}
    </LayoutColumn>
  </LayoutColumn>
);

const styles = StyleSheet.create({
  mailIcon: {
    marginTop: 7
  },
  phoneIcon: {
    marginTop: 6,
    marginLeft: 2
  }
});

export default ContactDetails;
