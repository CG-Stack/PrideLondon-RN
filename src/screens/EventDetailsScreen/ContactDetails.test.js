// @flow
import React from "react";
import { phonecall, email } from "react-native-communications";
import { shallow } from "enzyme";
import ContactDetails from "./ContactDetails";
import Touchable from "../../components/Touchable";
import text from "../../constants/text";

jest.mock("react-native-communications", () => ({
  phonecall: jest.fn(() => {}),
  email: jest.fn(() => {})
}));

beforeEach(() => {
  phonecall.mockClear();
  email.mockClear();
});

it("renders correctly", () => {
  const output = shallow(
    <ContactDetails email="milo@red-badger.com" phone="1-800-555-WOOF" />
  );
  expect(output).toMatchSnapshot();
});

it("calls the phone number on tap", () => {
  const output = shallow(
    <ContactDetails email="milo@red-badger.com" phone="1-800-555-WOOF" />
  );

  output
    .find(Touchable)
    .last()
    .simulate("press");
  expect(phonecall).toBeCalledWith("1-800-555-WOOF", false);
});

it("opens the configured email app on tap", () => {
  const output = shallow(
    <ContactDetails email="milo@red-badger.com" phone="1-800-555-WOOF" />
  );

  output
    .find(Touchable)
    .first()
    .simulate("press");
  expect(email).toBeCalledWith(
    ["milo@red-badger.com"],
    null,
    null,
    text.eventDetailsContactEmailSubject,
    text.eventeventDetailsContactEmailBody
  );
});
