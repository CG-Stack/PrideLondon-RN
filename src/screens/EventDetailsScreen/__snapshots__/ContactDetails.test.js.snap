// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    allowFontScaling={true}
    color="lightNavyBlueColor"
    markdown={false}
    onLayout={[Function]}
    style={
      Object {
        "marginBottom": 4,
        "marginTop": 8,
      }
    }
    type="h2"
  >
    Contact
  </Text>
  <View
    style={
      Object {
        "marginBottom": 8,
      }
    }
  >
    <IconItem
      icon={
        <Image
          source={1}
          style={
            Object {
              "marginTop": 7,
            }
          }
        />
      }
      style={Object {}}
    >
      <TextLink
        onPress={[Function]}
        style={Object {}}
        type="h4"
      >
        milo@red-badger.com
      </TextLink>
    </IconItem>
  </View>
  <View
    style={
      Object {
        "marginBottom": 8,
      }
    }
  >
    <IconItem
      icon={
        <Image
          source={1}
          style={
            Object {
              "marginLeft": 2,
              "marginTop": 6,
            }
          }
        />
      }
      style={Object {}}
    >
      <TextLink
        onPress={[Function]}
        style={Object {}}
        type="h4"
      >
        1-800-555-WOOF
      </TextLink>
    </IconItem>
  </View>
</View>
`;
